### PR TITLE
Refs 47: db:add preliminar entity relationship diagram

### PR DESCRIPTION
**Pull request by avisiedo**
This PR just add a draft for an entity-relationship to be reviewed.

Signed-off-by: Alejandro Visiedo <avisiedo@redhat.com>

### CODE REVIEW

File: docs/db-model.puml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+' avoid problems with angled crows feet
+skinparam linetype ortho
+
+' GORM custom types
+' https://gorm.io/docs/data_types.html
+
+' NOTES
+' - <<..>> syntax is not normalized, just a way to express different
+'   circunstances, as if the field is generated, or if the field is
```
[the remark this answers is not in the document]
Comment (avisiedo) [reply docs/db-model.puml]: thanks! updated

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-47

**Comment by jlsherrill** [comment]
⚠️ This task currently requires qe-approval, but this PR does not fully resolve the task.  Please contact QE to determine appropriate testing required.

**Comment by jlsherrill** [comment]
mentioned this on chat, but also wanted to record here, thoughts on using https://github.com/achiku/planter ?

**Comment by avisiedo** [comment]
I think is nice to use planter and keep the er-diagram in sync with the database schema! :)  I will add makefile rules to just run one command to get it! thanks

**Comment by rverdile** [comment]
+1, I like planter

File: mk/plantuml.mk [hunk 2]
```diff
@@ -4,13 +4,35 @@
...
+PLANTER=$(GO_OUTPUT)/planter
+
 .PHONY: plantuml-generate
 plantuml-generate: PLANTUML ?= $(shell command -v plantuml 2>/dev/null)
 plantuml-generate: PLANTUML ?= false
 plantuml-generate: $(patsubst docs/%.puml,docs/%.svg,$(wildcard docs/*.puml)) ## Generate diagrams
 
+.PHONY: docs/db-model.puml
+docs/db-model.puml: $(PLANTER)
+	$(GO_OUTPUT)/planter postgres://$(DATABASE_USER)@$(DATABASE_HOST)/$(DATABASE_NAME)?sslmode=disable -o $@
```
Comment (jlsherrill) [inline mk/plantuml.mk:22]: is DATABASE_PASSWORD not needed?  I'm getting an error:

```
/home/jlsherri/git/content-sources-backend/release/planter postgres://content@localhost/content?sslmode=disable -o docs/db-model.puml
2022/07/11 16:13:59 failed to load table def: pq: password authentication failed for user "content"
make: *** [mk/plantuml.mk:22: docs/db-model.puml] Error 1
```

**Comment by jlsherrill** [comment]
Would you want to add this to our docs?  we could either put it in our architecture doc:  https://github.com/content-services/content-sources-backend/blob/main/docs/architecture.md

or just link to the rendered image in the list here: https://github.com/content-services/content-sources-backend#more-info

**Comment by jlsherrill** [comment]
speaking of the rendered picture, wanna include that here too? :)

**Comment by avisiedo** [comment]
Sure! I paste here the current rendered diagram.

image

**Comment by avisiedo** [comment]
I think I have found a way that let to include the rendered diagram without to add the rendered image to the repository.

**Downsides**:
- It require the `.puml` file to exist into the repository.
- If the proxy service fails, the image is not rendered.
**Upside**: Allow to get the rendered diagram without upload the image file to the repository.

Below an example referencing the architecture diagram:

```raw
![Content Sources Architecturel](https://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/content-services/content-sources-backend/main/docs/architecture.puml)
```

Content Sources Database Model

**Comment by avisiedo** [comment]
I have updated a dynamic a reference that will render the diagram based on the `docs/db-model.puml` file at master branch; else, I can always add the `.png` file and link it.

It would render this: [dynamic image rendered](https://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/avisiedo/content-sources-backend/add-er-diagram/docs/db-model.puml).

Let me know your thoughts!

**Review by jlsherrill — APPROVED** [review]

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/14649800

**Pull request merged**